core/optaplanner-core-impl/src/main/java/org/optaplanner/core/impl/domain/variable/listener/support/PropertyChangedNotification.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2022 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.core.impl.domain.variable.listener.support;
+
+import org.optaplanner.core.api.domain.variable.VariableListener;
+import org.optaplanner.core.api.score.director.ScoreDirector;
+
+final class PropertyChangedNotification extends VariableListenerNotification {
+
+    PropertyChangedNotification(Object problemFact) {
+        super(problemFact);
+    }
```

Comment:
Since the property change is relevant to both problem fact and entity:

```suggestion
    PropertyChangedNotification(Object problemFactOrEntity) {
        super(problemFactOrEntity);
    }
```